pkg/kv/kvpb/api.go

```diff
@@ -1354,9 +1370,17 @@ func flagForLockStrength(l lock.Strength) flag {
 	return 0
 }
 
+func flagForIsReadOrIsWrite(l lock.Strength, durabilityType DurabilityType) flag {
+	if l == lock.None {
+		return isRead
+	}
+	return isWrite
```

Comment:
Should this consult the `durabilityType`?

Reply:
Oops, I was considering a different approach which I decided to backtrack on. Didn't press undo enough times :p